prow/config/branch_protection.go

```diff
@@ -241,15 +249,23 @@ func (c *Config) GetBranchProtection(org, repo, branch string) (*Policy, error)
 		return nil, err
 	}
 
-	return c.GetPolicy(org, repo, branch, *b)
+	return c.GetPolicy(githubClient, gitClient, org, repo, branch, *b)
 }
 
 // GetPolicy returns the protection policy for the branch, after merging in presubmits.
-func (c *Config) GetPolicy(org, repo, branch string, b Branch) (*Policy, error) {
+func (c *Config) GetPolicy(gitHubClient branchProtectionGitHubClient, gitClient *git.Client, org, repo, branch string, b Branch) (*Policy, error) {
 	policy := b.Policy
 
+	refGetter := func() (string, error) {
+		return gitHubClient.GetRef(org, repo, "heads/"+branch)
+	}
+	presubmits, err := c.GetPresubmits(gitClient, org+"/"+repo, refGetter)
```

Comment:
Wait so if we're only protecting the centrally-configured jobs and the synthetic job that's generated, why can't we use the static list here?

Reply:
So my understanding from what we discussed a couple of weeks back was that Cole suggested that we just have the Branchprotector get the jobs from the branch it should protect.

You both seem to not remember that so maybe my memories are wrong.

I personally have no strong opinions about this, but I don't see a technical reason why that shouldn't be possible. `inrepoconfig` must create dummy statuses if someone removes a context but other than that this should just work.

Reply:
Oh, fetching from the branch itself -- that sounds good but you are going to need to run this protector really often to make it consistent.

Reply:
Yep, true. So as mentioned, I personally don't really care. Should I keep it or should I remove it and we consider this a problem for another day?

Reply:
We've tried to move away from posting "Skipped" statuses so I favor avoiding that and relying on Tide to enforce that all the required jobs have run.  It seems wrong to apply branch protection policies across an entire repo when we know that some of the requirements won't actually apply to some of the PRs and that there will also be requirements missing for some PRs.

Reply:
Removed it